### PR TITLE
Fix type inference for fold

### PR DESCRIPTION
**Pull request by rajamony**

### CODE REVIEW

**Review by pierredebelen — CHANGES_REQUESTED** [review]
Could we please have a test in the compiler section instead of Query?

**Comment by rajamony** [comment]
Done. Test is simpler and now entirely in grammar

File: legend-engine-language-pure-compiler/src/main/java/org/finos/legend/engine/language/pure/compiler/toPureGraph/handlers/Handlers.java [hunk 1]
```diff
@@ -96,6 +96,21 @@ private static void updateTwoParamsLambda(Object lambda, GenericType newGenericT
...
+    {
+        if (lambda instanceof Lambda)
+        {
+            Variable variable = ((Lambda) lambda).parameters.get(0);
+            variable._class = PackageableElement.getUserPathForPackageableElement(newGenericType._rawType());
+            variable.multiplicity = m;
+
+            Variable variable2 = ((Lambda) lambda).parameters.get(1);
+            variable2._class = PackageableElement.getUserPathForPackageableElement(newGenericType2._rawType());
+            variable2.multiplicity = m;
```
Comment (pierredebelen) [inline legend-engine-language-pure-compiler/src/main/java/org/finos/legend/engine/language/pure/compiler/toPureGraph/handlers/Handlers.java:109]: The multiplicity of the lambda parameters should not be equal. The iterator will most likely be 1 but the accumulator should be equal to the one of the Lambda return.

**Review by pierredebelen — CHANGES_REQUESTED** [review]
Thanks for fixing the tests! Please look at the comment for the lambda parameters multiplicity inference.

File: legend-engine-language-pure-compiler/src/main/java/org/finos/legend/engine/language/pure/compiler/toPureGraph/handlers/Handlers.java (hunk 1 above)
Comment (rajamony) [reply legend-engine-language-pure-compiler/src/main/java/org/finos/legend/engine/language/pure/compiler/toPureGraph/handlers/Handlers.java]: Have to keep in mind it is type AND multiplicity! The only two-parameter lambda we now have with different types is fold, for which iterator is mult-1. Conversion from org.finos.legend.pure.m3.coreinstance.meta.pure.metamodel.multiplicity.Multiplicity to org.finos.legend.engine.protocol.pure.v1.model.packageableElement.domain.Multiplicity is inline. Not sure if there is a better way to do it. Please review.

Comment (pierredebelen) [reply legend-engine-language-pure-compiler/src/main/java/org/finos/legend/engine/language/pure/compiler/toPureGraph/handlers/Handlers.java]: Looks great now! Thanks for the update!

**Review by pierredebelen — APPROVED** [review]

**Pull request merged**